### PR TITLE
[#10987] feat(client-java): add view CRUD to RelationalCatalog

### LINKED ISSUE #10987: [Subtask] Add client-java view CRUD support for relational catalog
### Describe the subtask

Add view CRUD support to `client-java` relational catalog, including `listViews`, `loadView`, `createView`, `alterView`, and `dropView`, with robust namespace/path validation and edge-case unit tests.

### Parent issue

#604

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

This PR adds view CRUD support in `client-java` `RelationalCatalog` and includes a follow-up hardening patch:

- implement `ViewCatalog` operations: `listViews`, `loadView`, `createView`, `alterView`, and `dropView`
- convert view representations/columns between API objects and DTOs
- add explicit full-namespace validation for table/view request path formatting
- rename `getTableFullNamespace` to `getEntityFullNamespace` for clearer shared semantics
- add/extend unit tests for view happy paths and edge cases (null representations, invalid namespace)

### Why are the changes needed?

Java client needs first-class view management APIs for relational catalogs as part of the generic view support epic. The hardening changes also make invalid namespace inputs fail fast with clear errors.

Fix: #10987

### Does this PR introduce _any_ user-facing change?

Yes. `client-java` relational catalog now supports view CRUD operations through `ViewCatalog` APIs.

### How was this patch tested?

- Added and updated unit tests in `TestRelationalCatalogView`
- Ran:
  - `./gradlew :clients:client-java:test --tests org.apache.gravitino.client.TestRelationalCatalogView --no-daemon`
  - `./gradlew :clients:client-java:test --tests org.apache.gravitino.client.TestRelationalCatalog --tests org.apache.gravitino.client.TestRelationalCatalogView --no-daemon`

### CODE REVIEW

File: common/src/main/java/org/apache/gravitino/dto/rel/ViewDTO.java [hunk 1]
```diff
@@ -104,7 +106,23 @@ public Column[] columns() {
...
+                SQLRepresentationDTO dto = (SQLRepresentationDTO) rep;
+                return (Representation)
+                    SQLRepresentation.builder()
+                        .withDialect(dto.dialect())
+                        .withSql(dto.sql())
+                        .build();
+              }
+              return (Representation) rep;
+            })
+        .toArray(Representation[]::new);
```
[the remark this answers is not in the document]
Comment (mchades) [reply common/src/main/java/org/apache/gravitino/dto/rel/ViewDTO.java]: Updated in c81f38a3d: ViewDTO.representations() now delegates conversion to DTOConverters.fromDTOs(representations), centralizing conversion logic and removing per-call inline reconstruction.

File: clients/client-java/src/main/java/org/apache/gravitino/client/RelationalCatalog.java [hunk 2]
```diff
@@ -283,8 +300,169 @@ public boolean purgeTable(NameIdentifier ident) throws UnsupportedOperationExcep
...
+      String comment,
+      Column[] columns,
+      Representation[] representations,
+      String defaultCatalog,
+      String defaultSchema,
+      Map<String, String> properties)
+      throws NoSuchSchemaException, ViewAlreadyExistsException {
+    checkViewNameIdentifier(ident);
+
+    ColumnDTO[] columnDTOs = columns == null ? new ColumnDTO[0] : toDTOs(columns);
```
[the remark this answers is not in the document]
Comment (mchades) [reply clients/client-java/src/main/java/org/apache/gravitino/client/RelationalCatalog.java]: Updated in c81f38a3d: createView() now uses toDTOs(columns) directly, relying on DTOConverters null/empty handling for consistency with createTable().

File: clients/client-java/src/main/java/org/apache/gravitino/client/ErrorHandlers.java [hunk 3]
```diff
@@ -424,6 +435,61 @@ public void accept(ErrorResponse errorResponse) {
...
+
+      switch (errorResponse.getCode()) {
+        case ErrorConstants.ILLEGAL_ARGUMENTS_CODE:
+          throw new IllegalArgumentException(errorMessage);
+
+        case ErrorConstants.NOT_FOUND_CODE:
+          if (errorResponse.getType().equals(NoSuchSchemaException.class.getSimpleName())) {
+            throw new NoSuchSchemaException(errorMessage);
+          } else if (errorResponse.getType().equals(NoSuchCatalogException.class.getSimpleName())) {
+            throw new NoSuchCatalogException(errorMessage);
```
[the remark this answers is not in the document]
Comment (mchades) [reply clients/client-java/src/main/java/org/apache/gravitino/client/ErrorHandlers.java:455]: This doesn't need to be handled, as it remains consistent with the current TableErrorHandler's behavior.

**Comment by mchades** [comment]
@jerryshao could you plz help to review since comments from Copilot have been resolved, thx!

**Review by jerryshao — APPROVED** [review]
